zmon_aws_agent/main.py

```diff
@@ -0,0 +1,181 @@
+                resp = zmon_client.add_entity(entity)
+                logger.info('ZMon response ... {}'.format(resp.status_code))
+        except:
+            logger.exception('Failed to add entity!')
+
+    return new_entities
+
+
+def main():
+    argp = argparse.ArgumentParser(description='ZMon AWS Agent')
+    argp.add_argument('-e', '--entity-service', dest='entityservice')
+    argp.add_argument('-r', '--region', dest='region', default=None)
+    argp.add_argument('-j', '--json', dest='json', action='store_true')
+    argp.add_argument('--no-oauth2', dest='disable_oauth2', action='store_true', default=False)
+    args = argp.parse_args()
+
+    if not args.disable_oauth2:
+        tokens.configure()
+        tokens.manage('uid', ['uid'])
+        tokens.start()
+
+    logging.basicConfig(level=logging.INFO)
+
+    # 1. Determine region
+    if not args.region:
+        logger.info('Trying to figure out region..')
+        try:
+            response = requests.get('http://169.254.169.254/latest/meta-data/placement/availability-zone', timeout=2)
+        except:
+            logger.error('Region was not specified as a parameter and can not be fetched from instance meta-data!')
+            raise
+        region = response.text[:-1]
+    else:
+        region = args.region
+
+    logger.info('Using region: {}'.format(region))
+
+    logger.info('Entity service URL: %s', args.entityservice)
+
+    logger.info('Reading DNS data for hosted zones')
+    aws.populate_dns_data()
+
+    # 2. Get running apps
+    apps = aws.get_running_apps(region)
+
+    infrastructure_account = None
+
+    elbs = []
+    scaling_groups = []
+    rds = []
+    elasticaches = []
+    dynamodbs = []
+
+    new_entities = []
+    to_be_removed = []
+
+    if len(apps) > 0:
+        infrastructure_account = apps[0]['infrastructure_account']
+        elbs = aws.get_running_elbs(region, infrastructure_account)
+        scaling_groups = aws.get_auto_scaling_groups(region, infrastructure_account)
+        rds = aws.get_rds_instances(region, infrastructure_account)
+        elasticaches = aws.get_elasticache_nodes(region, infrastructure_account)
+        dynamodbs = aws.get_dynamodb_tables(region, infrastructure_account)
+
+    if infrastructure_account:
+        account_alias = aws.get_account_alias(region)
+        ia_entity = {
+            'type': 'local',
+            'infrastructure_account': infrastructure_account,
+            'account_alias': account_alias,
+            'region': region,
+            'id': 'aws-ac[{}:{}]'.format(infrastructure_account, region),
+            'created_by': 'agent',
+        }
+
+        application_entities = aws.get_apps_from_entities(apps, infrastructure_account, region)
+
+        current_entities = elbs + scaling_groups + apps + application_entities + rds + elasticaches + dynamodbs
+
+        # 3. ZMON entities
+        token = None if args.disable_oauth2 else os.getenv('ZMON_AGENT_TOKEN', tokens.get('uid'))
```

Comment:
you don't need `ZMON_AGENT_TOKEN` --- the Python tokens library already allows setting tokens manually https://github.com/zalando-stups/python-tokens